pkg/experimental/pkg_filegroup.bzl

```diff
@@ -578,6 +591,83 @@ pkg_mkdirs = rule(
             The default is `dir`.
             """,
             default = "dir",
+            # TODO(nacl): use "values" here
+        ),
+    },
+)
+
+def _pkg_mklinks_impl(ctx):
+    _validate_attr(ctx.attr.attrs)
+    return [
+        PackageSymlinkInfo(
+            link_map = ctx.attr.links,
+            attrs = ctx.attr.attrs,
+            section = ctx.attr.section,
+        ),
+    ]
+
+pkg_mklinks = rule(
+    doc = """pkg_filegroup-like-rule for symlink management.
+
+    This rule results in the creation of one or more symbolic links a package.
+    The link may point to a location outside of it.
+
+    """,
+    implementation = _pkg_mklinks_impl,
+    attrs = {
+        "links": attr.string_dict(
+            doc = """Link mappings to create.
+
+            The keys of this dict are the link names, the values are the source
+            files.
+
+            Note that relative paths are important here: the package builder MAY
```

Comment:
I don't understand how you are applying "relative" here. 
Let's say you have a package  with a subfolder "data" (no BUILD file, so it is in the same package)

pkg_mklinks(links = {"bar":  "data/bar.dat"})

That should obviously be allowed.  What if data/bar.dat is actually another package, but not a dep of some other rule in this one. Is that allowed?

Could it be links = {"baz/bar":  "whatever"}?
Where baz does not exist anywhere in the source tree?

Reply:
> I don't understand how you are applying "relative" here.

The issue at hand is that RPM packages only support defining paths relative to a root.  For example, if you say, have a mapping like this:

```
pkg_mklinks(links = {"bar": "data/bar.dat"})
```

This creates a symbolic link within the package named `bar` pointing to a file named `data/bar.dat`.  No checks are made to confirm whether `data/bar.dat` exists; the link is simply created.

Tar archives give you the option of specifying a path relative to the directory in which it's being extracted, or absolute relative to the UNIX filesystem root.  RPM packages are limited to files specified relative to the filesystem root, although [relocatable packages](http://ftp.rpm.org/max-rpm/ch-rpm-reloc.html) are possible to create.  However, AFAICT, paths in `%files` must still be specified relative to the filesystem root (or in this case, the relocation prefix root).

> Let's say you have a package with a subfolder "data" (no BUILD file, so it is in the same package)
>
>pkg_mklinks(links = {"bar": "data/bar.dat"})
>
> That should obviously be allowed. What if data/bar.dat is actually another package, but not a dep of some other rule in this one. Is that allowed?

This behavior is supported, yes.  It's tested as a part of the unit test updates added along with this pull request.

> Could it be links = {"baz/bar": "whatever"}?
> Where baz does not exist anywhere in the source tree?

Absolutely.  One of the nitty gritty details of this design is the implicit directory structures created by the various `pkg_filegroup`-like rules.  This will have to be thought about.

Regardless, my takeaway from this is that the documentation needs to be made more clear.  I'll adjust the documentation to make it clearer that we're explicitly talking about paths within packages.

Reply:
Documentation updates done.

I decided to throw out mentions of how relative paths are controlled in favor of documenting it more comprehensively as a part of #174.